enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,787 @@
+  resources = {
+    "cpu": "",
+    "cpuset": "0-1",
+  }
+```
+
+The `annotation` field is used to match pods that should be treated as
+having the workload type. The annotation key on the pod is compared
+for an exact match against the value specified in the configuration
+file.
+
+The `resource_annotation_prefix` is the start of the annotation key
+used to pass settings from the admission hook down to CRI-O.
+
+The `resources` map associates annotation suffixes with default
+values. CRI-O will define a well-known set of resources and other
+values will not be allowed. In OpenShift, we do not want to allow
+arbitrary pods to set their own `cpuset` but other CRI-O users may
+want that ability.
+
+To pass a setting into CRI-O, the pod should have an annotation made
+by combining the `resource_annotation_prefix`, the key from
+`resources`, and the container name, like this:
+
+```text
+io.openshift.workload.management.cpu/container_name = {value}
+```
+
+In the management workload case, we will configure it with values like
+
+```ini
+[crio.runtime.workloads.management]
+  annotation = "workload.openshift.io/management"
+  resource_annotation_prefix = "io.openshift.workload.management"
+  resources = {
+    "cpu": "",
+    "cpuset": "0-1",
+  }
+```
+
+CRI-O will be configured to support a new annotation on pods,
+`io.openshift.workload.management.cpu/{container-name}`.
+
+```ini
+[crio.runtime.runtimes.runc]
+  runtime_path = "/usr/bin/runc"
+  runtime_type = "oci"
+```
+
+Pods that have the `io.openshift.workload.management.cpu` annotation
+will have their cpuset configured to the value from the appropriate
+workload configuration, as well as have their CPU shares configured to
+the value of the annotation.
+
+Note that this field does not conflict with the `infra_ctr_cpuset`
+config option, as the infra container will still be put in that
+cpuset.  They can be configured as the same value if the infra
+container should also be considered to be managed.
+
+#### API Server Admission Hook
+
+A new admission hook in the kubernetes API server within OpenShift
+will mutate pods when they are created to make 3 changes.
+
+1. It will move CPU requests to workload CPU requests, so that
+   the scheduler can successfully place the pod on the node, even
+   though the sum of the CPU requests for all management workloads may
+   exceed the actual CPU capacity of the management CPU pool.
+2. It will strip all container-specific workload annotations from new
+   pods. This both prevents a user from configuring a pod incorrectly
+   and prevents unprivileged users from running their pods in a CPU
+   pool they should not have access to.
+3. It will add annotations with for each container with the value
+   equal to the original CPU requests * 1000, so that CRI-O can use
+   the value to configure the CPU shares for the container.
+
+We will not change pods in a way that changes their quality-of-service
```

Comment:
nit: Can use the upstream terminology here: `Quality of Service (QoS)`. See https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/.